clientv3/retry.go

```diff
@@ -54,14 +63,50 @@ func (c *Client) newRetryWrapper() retryRpcFunc {
 	}
 }
 
+func (c *Client) newRdonlyRetryWrapper() retryRpcFunc {
+	return func(rpcCtx context.Context, f rpcFunc) error {
+		for {
+			err := f(rpcCtx)
+			if err == nil {
+				return nil
+			}
+
+			// always stop retry on etcd errors other than invalid auth token
+			eErr := rpctypes.Error(err)
+			if etcdErr, ok := eErr.(rpctypes.EtcdError); ok {
+				if etcdErr.Code() == rpctypes.ErrInvalidAuthToken.(rpctypes.EtcdError).Code() {
+					gterr := c.getToken(rpcCtx)
+					if gterr != nil {
+						return err // return the original error for simplicity
+					}
+					continue
+
+				}
+				return err
+			}
+
+			return err
+		}
+	}
+}
+
 type retryKVClient struct {
 	pb.KVClient
-	retryf retryRpcFunc
+	retryf       retryRpcFunc
+	rdonlyRetryf retryRpcFunc
```

Comment:
I see. Then the new `retryWriteKVClient` is hidden from clients? Do we need to provide a configurable option for using it by the clients? I think just using the new `retryKvClient` in a fixed manner would be ok, though.